zqd/handlers.go

```diff
@@ -424,60 +424,30 @@ func handleLogPost(c *Core, w http.ResponseWriter, r *http.Request) {
 	}
 	defer cancel()
 
-	var req api.LogPostRequest
-	if !request(c, w, r, &req) {
-		return
-	}
-	if len(req.Paths) == 0 {
-		respondError(c, w, r, zqe.E(zqe.Invalid, "empty paths"))
-		return
-	}
-	op, err := ingest.NewLogOp(ctx, s.Storage(), req)
+	form, err := r.MultipartReader()
 	if err != nil {
-		respondError(c, w, r, err)
-		return
-	}
-	w.Header().Set("Content-Type", "application/ndjson")
-	w.WriteHeader(http.StatusAccepted)
-	logger := c.requestLogger(r)
-	pipe := api.NewJSONPipe(w)
-	if err := pipe.SendStart(0); err != nil {
-		logger.Warn("error sending payload", zap.Error(err))
+		respondError(c, w, r, zqe.ErrInvalid(err))
 		return
 	}
-	ticker := time.NewTicker(time.Second)
-	defer ticker.Stop()
-loop:
-	for {
-		select {
-		case warning, ok := <-op.Status():
-			if !ok {
-				break loop
-			}
-			err := pipe.Send(api.LogPostWarning{
-				Type:    "LogPostWarning",
-				Warning: warning,
-			})
-			if err != nil {
-				logger.Warn("error sending payload", zap.Error(err))
-				return
-			}
-		case <-ticker.C:
-			if err := pipe.Send(op.Stats()); err != nil {
-				logger.Warn("error sending payload", zap.Error(err))
-				return
-			}
-		}
-	}
-	// send final status
-	if err := pipe.Send(op.Stats()); err != nil {
-		logger.Warn("error sending payload", zap.Error(err))
-		return
+
+	store := s.Storage()
```

Comment:
NIt: Declare this closer to its use or, better yet, not at all (by saying `s.Storage().Write` below).